rtengine/clutstore.cc

```diff
@@ -3,440 +3,233 @@
-                resultImg->g( x, y ) = step * g;
-                resultImg->b( x, y ) = step * b;
-                ++pos;
-            }
-        }
+    clut_image = loadFile(filename, "", clut_level);
+    Glib::ustring name, ext;
+    splitClutFilename(filename, name, ext, clut_profile);
+
+    if (clut_image) {
+        clut_filename = filename;
+        return true;
     }
 
-    return resultImg;
+    return false;
 }
 
-
-bool HaldCLUT::isValid() const
+rtengine::HaldCLUT::operator bool() const
 {
-    return m_clutImage != 0;
+    return static_cast<bool>(clut_image);
 }
 
-void HaldCLUT::getRGB( float rr, float gg, float bb, float &outR, float &outG, float &outB ) const
+Glib::ustring rtengine::HaldCLUT::getFilename() const
 {
-    rr /= MAXVALF;
-    gg /= MAXVALF;
-    bb /= MAXVALF;
-    correct( *m_clutImage, m_level, rr, gg, bb, outR, outG, outB );
+    return clut_filename;
 }
 
-inline float valF( unsigned char val )
+Glib::ustring rtengine::HaldCLUT::getProfile() const
 {
-    return float( val ) / MAXVAL8;
+    return clut_profile;
 }
 
-// copypasted from http://www.quelsolaar.com/technology/clut.html
-void HaldCLUT::correct( const HaldCLUT::RawClut& clut, int level, float rr, float gg, float bb, float &outR, float &outG, float &outB )
+void rtengine::HaldCLUT::getRGB(float r, float g, float b, float& out_r, float& out_g, float& out_b) const
 {
-    int color, red, green, blue, i, j;
-    float tmp[6], r, g, b;
-    level =  level * level;
+    const unsigned int level = clut_level * clut_level;
 
-    red = rr * (float)(level - 1);
+    const float flevel_minus_one = static_cast<float>(level - 1) / 65535.0f;
+    const float flevel_minus_two = static_cast<float>(level - 2);
 
-    if(red > level - 2) {
-        red = (float)level - 2;
-    }
+    const unsigned int red = std::min(flevel_minus_two, r * flevel_minus_one);
+    const unsigned int green = std::min(flevel_minus_two, g * flevel_minus_one);
+    const unsigned int blue = std::min(flevel_minus_two, b * flevel_minus_one);
 
-    if(red < 0) {
-        red = 0;
-    }
+    r = r * flevel_minus_one - red;
+    g = g * flevel_minus_one - green;
+    b = b * flevel_minus_one - blue;
 
-    green = gg * (float)(level - 1);
+    const unsigned int level_square = level * level;
 
-    if(green > level - 2) {
-        green = (float)level - 2;
-    }
+    const unsigned int color = red + green * level + blue * level_square;
 
-    if(green < 0) {
-        green = 0;
-    }
-
-    blue = bb * (float)(level - 1);
-
-    if(blue > level - 2) {
-        blue = (float)level - 2;
-    }
-
-    if(blue < 0) {
-        blue = 0;
-    }
-
-    r = rr * (float)(level - 1) - red;
-    g = gg * (float)(level - 1) - green;
-    b = bb * (float)(level - 1) - blue;
+    unsigned int x[2];
+    unsigned int y[2];
+    posToXy(color, clut_image->getWidth(), x, y);
 
-    color = red + green * level + blue * level * level;
+    float tmp1[4] __attribute__((aligned(16)));
```

Comment:
please use ALIGNED16 instead of __attribute__((aligned(16)))
ALIGNED16 is defined in rtengine/opthelper.h, so you may need to include that file